pkg/bloomgateway/util.go

```diff
@@ -36,12 +38,22 @@ func getFromThrough(refs []*logproto.ShortRef) (model.Time, model.Time) {
 
 // convertToSearches converts a list of line filter expressions to a list of
 // byte slices that can be used with the bloom filters.
-// TODO(chaudum): Currently this function only supports equality matchers,
-// but we eventually also want to support regex matchers.
-func convertToSearches(filters []*logproto.LineFilterExpression) [][]byte {
-	searches := make([][]byte, 0, len(filters))
+// TODO(chaudum): Tokenize filter strings
+func convertToSearches(cfg tokenSettings, filters []syntax.LineFilter) [][]byte {
+	// never use a skip factor for search n-grams
+	t := v1.NewNGramTokenizer(cfg.nGramLen, 0)
+	// Can we assume an avg filter string length of 12 chars?
+	// Then the capacity would be (len(filterString) - (nGramLen-1)) * numFilters
+	searches := make([][]byte, 0, (13-cfg.nGramLen)*len(filters))
 	for _, f := range filters {
-		searches = append(searches, []byte(f.Match))
+		if f.Ty == labels.MatchEqual {
+			it := t.Tokens(f.Match)
+			for it.Next() {
+				key := make([]byte, cfg.nGramLen)
+				_ = copy(key, it.At())
```

Comment:
Shouldn't this work or will you get a linter error?
```suggestion
				copy(key, it.At())
```

Reply:
Linter says "No!" :D